### PR TITLE
Lint with gofmt 1.19, upgrade to go 1.19

### PR DESCRIPTION
**Pull request by dreamsmasher**
## Description

Just a housekeeping task since we've been putting off switching to gofmt 1.19 for a while. Most formatting is preserved, save for trailing line comments. Only 1 doc comment had to be manually edited in runtime/sema/type.go (IsSubType).

______

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/sema/type.go [hunk 1]
```diff
@@ -4982,15 +4968,22 @@ func (t *AddressType) GetMembers() map[string]MemberResolver {
...
+// of parameter type R? This is the more frequent use-case.
+//
+//	IsSameTypeKind()
+//
+// To check if a type strictly belongs to a certain category. e.g: Is the
+// expression type T is any of the integer types, but nothing else.
+// Another way to check is, asking the question of "if the subType is Never,
+// should the check still pass?". A common code-smell for potential incorrect
+// usage is, using IsSubType() method with a constant/pre-defined superType.
+// e.g: IsSubType(<<someType>>, FixedPointType)
```
Comment (dreamsmasher) [inline runtime/sema/type.go:4986]: specifically here, this now renders as code blocks followed by description paragraphs.

Comment (SupunS) [inline runtime/sema/type.go:4986]: The default automatic formatting for this one seems ok to me:
```suggestion
//   - IsSubType()      - To check the assignability. e.g: Is argument type T is a sub-type
//     of parameter type R. This is the more frequent use-case.
//   - IsSameTypeKind() - To check if a type strictly belongs to a certain category. e.g: Is the
//     expression type T is any of the integer types, but nothing else.
//     Another way to check is, asking the question of "if the subType is Never,
//     should the check still pass?". A common code-smell for potential incorrect
//     usage is, using IsSubType() method with a constant/pre-defined superType.
//     e.g: IsSubType(<<someType>>, FixedPointType)
```

**Review by SupunS — COMMENTED** [review]
Thank you for cleaning these up. It was a bit annoying when gofmt updated random places in random PRs. 🙏

File: runtime/sema/type.go (hunk 1 above)
Comment (dreamsmasher) [reply runtime/sema/type.go:4986]: @turbolent thoughts? this was one example you specifically cited as being broken in a previous PR formatted with gofmt 1.19

**Comment by dreamsmasher** [comment]
need to upgrade to go 1.19 because of gofmt.

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/onflow/cadence
 
-go 1.18
+go 1.19
```
Comment (turbolent) [inline go.mod:3]: This indicates the minimum version of Go. We should be able to keep requiring Go 1.18

```suggestion
go 1.18
```

File: runtime/sema/type.go (hunk 1 above)
Comment (turbolent) [inline runtime/sema/type.go:4986]: Is it possible to indent the descriptions under the function names? 

Not tested, but just to show what I mean:

```suggestion
//	* IsSubType():
//
//     To check the assignability, e.g: is argument type T is a sub-type
//     of parameter type R? This is the more frequent use-case.
//
//	* IsSameTypeKind():
//
//     To check if a type strictly belongs to a certain category. e.g: Is the
//     expression type T is any of the integer types, but nothing else.
//     Another way to check is, asking the question of "if the subType is Never,
//     should the check still pass?". A common code-smell for potential incorrect
//     usage is, using IsSubType() method with a constant/pre-defined superType.
//     e.g: IsSubType(<<someType>>, FixedPointType)
```

**Review by turbolent — COMMENTED** [review]
LGTM, just one comment we could maybe improve

**Comment by turbolent** [comment]
Only CI should be updated, the packages themselves should stay at 1.18

File: runtime/sema/type.go (hunk 1 above)
Comment (turbolent) [reply runtime/sema/type.go:4986]: Ah, I missed this thread while reviewing, but had also commented on it here: https://github.com/onflow/cadence/pull/2041#discussion_r989524751

File: go.mod (hunk 2 above)
Comment (dreamsmasher) [reply go.mod:3]: wouldn't that create issues for contributors since they'll fail ci-lint?

File: runtime/sema/type.go (hunk 1 above)
Comment (dreamsmasher) [reply runtime/sema/type.go:4986]: Ah yeah, but the asterisks are replaced by hyphens after formatting.

**Review by turbolent — APPROVED** [review]
Nice!

**Review by SupunS — APPROVED** [review]

**Comment by dreamsmasher** [comment]
YESSSSS

**Pull request merged**